### PR TITLE
[SYSE-337]: Update templates and Dr.Releng to support release-4-lts

### PR DESCRIPTION
**Pull request by asutosh**
- Disables api-tests within release workflow for release-4-lts.
- Uses a feature named `auto-test` to implement this, all the supported repos and branches need to have this feature set in the config to render api-tests.

### CODE REVIEW

File: policy/templates/releng/.github/workflows/release.yml [hunk 1]
```diff
@@ -26,95 +26,11 @@ env:
...
-      matrix:
-        conf: {{`${{ fromJson(needs.test-controller-api.outputs.conf) }}`}}
-        db: {{`${{ fromJson(needs.test-controller-api.outputs.db) }}`}}
-        {{ if ne .Name "tyk-pump" -}} pump: {{`${{ fromJson(needs.test-controller-api.outputs.pump) }}`}} {{ end }} {{/* pump array not needed when the repo is pump */}}
-        {{ if ne .Name "tyk-sink" -}} sink: {{`${{ fromJson(needs.test-controller-api.outputs.sink) }}`}} {{ end }}
-        include:
-          - db: postgres15
-            markers: "and not sql"
-{{- template "api-tests" . }}
+{{- if has "api-tests-suit" .Branchvals.Features }}
```
Comment (alephnull) [inline policy/templates/releng/.github/workflows/release.yml:30]: Suggest calling this feature `auto-test` as it contains UI too and more in the future.

File: policy/templates/releng/.github/workflows/release.yml [hunk 2]
```diff
@@ -26,95 +26,11 @@ env:
...
+  {{- template "test-controller" . }}
+  {{- template "api-test-init" . }}
```
Comment (alephnull) [inline policy/templates/releng/.github/workflows/release.yml:32]: Can this not be merged into the `test-controller` template?

**Review by alephnull — COMMENTED** [review]
Minor changes.

File: policy/templates/releng/.github/workflows/release.yml (hunk 2 above)
Comment (asutosh) [reply policy/templates/releng/.github/workflows/release.yml:32]: ah yes, makes sense to merge them together. Will update

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -67,6 +67,10 @@ jobs:
             branch: release-5-lts
```
Comment (alephnull) [inline .github/workflows/release.yml:67]: Doesn't making this `[ release-5-lts, release-4-lts ]` work?

File: policy/templates/releng/.github/workflows/release.yml [hunk 4]
```diff
@@ -26,95 +26,10 @@ env:
...
-        db: {{`${{ fromJson(needs.test-controller-api.outputs.db) }}`}}
-        {{ if ne .Name "tyk-pump" -}} pump: {{`${{ fromJson(needs.test-controller-api.outputs.pump) }}`}} {{ end }} {{/* pump array not needed when the repo is pump */}}
-        {{ if ne .Name "tyk-sink" -}} sink: {{`${{ fromJson(needs.test-controller-api.outputs.sink) }}`}} {{ end }}
-        include:
-          - db: postgres15
-            markers: "and not sql"
-{{- template "api-tests" . }}
+{{- if has "auto-test" .Branchvals.Features }}
+  {{- template "test-controller" . }}
+  {{- template "api-tests" . }}
```
Comment (alephnull) [inline policy/templates/releng/.github/workflows/release.yml:32]: Rename the `test-controller` template to `auto-test` (in line with the feature).
And move this line into `auto-test`? This gives the impression that only api-tests are run, while all auto tests are run.

**Review by alephnull — CHANGES_REQUESTED** [review]
Better get this merged before @ermirizio lands his changes or you'll have to rework this >:-)

**Review by alephnull — APPROVED** [review]

**Pull request merged**